lambda-buffers-compiler/src/LambdaBuffers/Compiler/TypeClassCheck/Utils.hs

```diff
@@ -276,29 +275,30 @@ buildClasses cis = foldM go M.empty (concat $ M.elems cis)
 
 type Instances = S.Set Instance
 
-getInstances :: M.Map FQClassName Class -> P.ModuleName -> [P.InstanceClause] -> Either TypeClassError Instances
+getInstances :: M.Map FQClassName Class -> P.ModuleName -> [P.DeriveClause] -> Either TypeClassError Instances
 getInstances ctable mn = foldM go S.empty
   where
-    go :: S.Set Instance -> P.InstanceClause -> Either TypeClassError Instances
-    go acc (P.InstanceClause cn h csts si') = case ctable ^? ix cref of
+    go :: S.Set Instance -> P.DeriveClause -> Either TypeClassError Instances
+    go acc (P.DeriveClause cn h si') = case ctable ^? ix cref of
       Nothing -> throwError $ UnknownClass cref si'
       Just cls -> do
         let p = tyToPat h
-        cs <- traverse goConstraint csts
-        let inst = C cls p :<= cs
+        let inst = C cls p :<= []
         checkInstance inst
         pure $ S.insert inst acc
```

Comment:
I think you need to adjust the naming at this point as it's getting confusing.

Imo you should internally use Rule? As `Instance` is sort of a Rule that a User can define, which is more restricted.